woocommerce/woocommerce-ios
# [Mobile Payments] Parse and rename eligibility flag

Closes #4160 

⚠️ This PR is against the feature branch implementing support for Card Present Payments, not against develop ⚠️


There is ongoing work to pass down the eligibility flag from WCPay; https://github.com/Automattic/woocommerce-payments-server/pull/813

To be completely honest, I am not sure if submitting this now is actually a good thing, give we'd still need to come back to this in #4241 . My thinking was that it's better to get a bit closer to the final stage now and leave one line (in #4241) for later. But who knows.

The feature flag is hardcoded to true for now, but I've logged #4241 to revert that as soon as https://github.com/Automattic/woocommerce-payments-server/pull/813 is available

## Changes
* Parse the eligibility flag using the proper key, to integrate with the work done on the backend.
* Rename the eligibility flag to match the name it has on the backend.

I excluded unit tests, will add those with #4241 

## How to test
Testing requires:

1. Setting up a store for Card Present Payment Testing (also P91TBi-4BH-p2 for more specific instructions)
2. WCPay 2.4.0

* Checkout the branch
* Connect to a reader
* Navigate to an eligible order. The "Collect Payment" button should be available in debug builds
* Navigate to Settings. The "Manage Card Readers" section should be available in debug builds.

Update release notes:
